### PR TITLE
Explained Checks-Effects-Interactions

### PR DESCRIPTION
**Pull request by leonardoalt**
Adaptation of https://github.com/ethereum/solidity/pull/13215

### CODE REVIEW

File: docs/security-considerations.rst [hunk 1]
```diff
@@ -116,6 +116,13 @@ outlined further below:
         }
     }
 
+The Checks-Effects-Interactions pattern ensures that all code paths through a contract complete all required verification
```
Comment (Marenz) [inline docs/security-considerations.rst]:
```suggestion
The Checks-Effects-Interactions pattern ensures that all code paths through a contract complete all required verifications
```

File: docs/security-considerations.rst [hunk 2]
```diff
@@ -116,6 +116,13 @@ outlined further below:
...
+of supplied parameters (Checks) before modifying contract state; then makes any changes to contract state (Effects);
```
Comment (Marenz) [inline docs/security-considerations.rst]:
```suggestion
of supplied parameters (Checks) before modifying the contract state; only then it makes any changes to contract state (Effects);
```

**Review by Marenz — CHANGES_REQUESTED** [review]
I am really not happy with the English in the whole paragraph. I tried rephrasing/rewriting it, but I am not in the right mental state for that now (super tired) and I don't actually know the pattern

**Comment by leonardoalt** [comment]
made some changes

File: docs/security-considerations.rst [hunk 3]
```diff
@@ -116,6 +116,13 @@ outlined further below:
         }
     }
 
+The Checks-Effects-Interactions pattern ensures that all code paths through a contract complete all required checks
+of supplied parameters before modifying the contract's state (Checks); only then it makes any changes to the state (Effects);
```
Comment (Marenz) [inline docs/security-considerations.rst]:
```suggestion
of the supplied parameters before modifying the contract's state (Checks); only then it makes any changes to the state (Effects);
```

**Review by Marenz — CHANGES_REQUESTED** [review]

**Review by Marenz — APPROVED** [review]

**Pull request merged**